crates/brainfuck_prover/src/components/instruction/table.rs

```diff
@@ -83,11 +85,76 @@ impl InstructionTable {
     pub fn get_row(&self, row: &InstructionTableRow) -> Option<&InstructionTableRow> {
         self.table.iter().find(|r| *r == row)
     }
+
+    /// Retrieves the last row in the Instruction Table.
+    ///
+    /// # Returns
+    /// An `Option` containing a reference to the last [`InstructionTableRow`] in the table,
+    /// or `None` if the table is empty.
+    pub fn last_row(&self) -> Option<&InstructionTableRow> {
+        self.table.last()
+    }
+}
+
+impl From<Vec<Registers>> for InstructionTable {
+    fn from(registers: Vec<Registers>) -> Self {
+        // Create a mutable copy of the input `registers` to sort it.
+        let mut sorted_registers = registers;
+        // Sort `sorted_registers` by
+        // 1. `ip` (instruction pointer)
+        // 2. `clk` (clock cycle)
+        sorted_registers.sort_by(|a, b| a.ip.cmp(&b.ip).then_with(|| a.clk.cmp(&b.clk)));
+
+        // Initialize a new instruction table to store the sorted and processed rows.
+        let mut instruction_table = Self::new();
+
+        // Use `current_ip` to track the current instruction pointer and group elements.
+        let mut current_ip = None;
+
+        for register in &sorted_registers {
```

Comment:
Extract the required registers, collect them in a Vec and call `InstructionTable::add_rows` ?

Reply:
Or simply with the method `add_from_registers`